mobile_app_connector/models/payment_transaction.py

```diff
@@ -0,0 +1,57 @@
+# coding: utf-8
+
+import logging
+from odoo import api, models, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    @api.multi
+    def cancel_transaction(self):
+        """
+        Called by ir_action_rule in order to cancel the transaction that was
+        not updated after a while.
+        :return: True
+        """
+        for transaction in self:
+            if transaction.invoice_id.origin in ["ios", "android"]:
+                """
+                Aborted transaction from the app. We must keep sponsorships but can
+                clean additional gifts
+                """
+                inv = transaction.invoice_id
+                inv.action_invoice_cancel()
+                inv.action_invoice_draft()
```

Comment:
I would move this below the `if` at line 32, to let the invoices cancelled in the other case.